examples/webgpu_mrt_readback.html

```diff
@@ -0,0 +1,256 @@
+				renderer.setAnimationLoop( render );
+				document.body.appendChild( renderer.domElement );
+
+				// Create a multi render target with Float buffers
+
+				renderTarget = new THREE.RenderTarget(
+					window.innerWidth * window.devicePixelRatio,
+					window.innerHeight * window.devicePixelRatio,
+					{ count: 2, minFilter: THREE.NearestFilter, magFilter: THREE.NearestFilter }
+				);
+
+				// Name our G-Buffer attachments for debugging
+
+				renderTarget.textures[ 0 ].name = 'diffuse';
+				renderTarget.textures[ 1 ].name = 'normal';
+
+
+				// Init readback render target
+				//Be careful with the size! 512 is already big. Reading data back from the GPU is computationally intensive
+				size = 512;	
+				readbackTarget = new THREE.RenderTarget(size, size, { count: 2 });
+
+				// Scene setup
+
+				scene = new THREE.Scene();
+				scene.background = new THREE.Color( 0x222222 );
+
+				camera = new THREE.PerspectiveCamera( 70, window.innerWidth / window.innerHeight, 0.1, 50 );
+				camera.position.z = 4;
+
+
+				const loader = new THREE.TextureLoader();
+
+				const diffuse = loader.load( 'textures/hardwood2_diffuse.jpg' );
+				diffuse.colorSpace = THREE.SRGBColorSpace;
+				diffuse.wrapS = THREE.RepeatWrapping;
+				diffuse.wrapT = THREE.RepeatWrapping;
+
+				torus = new THREE.Mesh(
+					new THREE.TorusKnotGeometry( 1, 0.3, 128, 32 ),
+					new WriteGBufferMaterial( diffuse )
+				);
+
+				scene.add( torus );
+
+
+				material = new ReadGBufferMaterial( renderTarget.textures[ 0 ], renderTarget.textures[ 1 ] )
+				quadMesh = new QuadMesh( material );
+
+				// Controls
+
+				new OrbitControls( camera, renderer.domElement );
+
+				window.addEventListener( 'resize', onWindowResize );
+
+			}
+
+			function onWindowResize() {
+
+				camera.aspect = window.innerWidth / window.innerHeight;
+				camera.updateProjectionMatrix();
+
+				renderer.setSize( window.innerWidth, window.innerHeight );
+
+				const dpr = renderer.getPixelRatio();
+				renderTarget.setSize( window.innerWidth * dpr, window.innerHeight * dpr );
+
+			}
+
+			async function render( time ) {
+
+				torus.rotation.y = ( time / 1000 ) * .4;
+
+				// render scene into target
+				renderer.setRenderTarget( renderTarget );
+				renderer.render( scene, camera );
+
+				// render post FX
+				renderer.setRenderTarget( null );
+
+				if(selector == 1){
+					quadMesh.material = material;
+					quadMesh.render( renderer );
+				}
+
+				await readback();
+			}
+
+
+			async function readback(){
+
+				renderer.setRenderTarget( readbackTarget );
+				renderer.render( scene, camera );
+				renderer.setRenderTarget( null );
+
+				const width = readbackTarget.width;
+				const height = readbackTarget.height;
+
+				if(selector == 2){
+					const pixelBuffer0 = await renderer.readRenderTargetPixelsAsync(readbackTarget, 0, 0, width, height, 0);	//zero is optional
+					
+					const pixelBufferTexture0 = new THREE.DataTexture(pixelBuffer0, width, height);
```

Comment:
Seems that `DataTexture` and `Material` are created in each frame too

Reply:
Ah, something else slipped through my fingers. Yes, of course that shouldn't be the case

Reply:
Now it looking good. Have you meanwhile seen the readback_Normal with taking #28197 into account?